core/src/container.cpp

```diff
@@ -109,20 +109,18 @@ void ContainerBasePrivate::compute() {
 template <Interface::Direction dir>
 void ContainerBasePrivate::copyState(Interface::iterator external, const InterfacePtr& target, bool updated) {
 	if (updated) {
-		// TODO(v4hn): This is inefficient, consider storing inverse mapping for each start/end pair in children
-		// for parallel containers there will be multiple internal states
-		std::for_each(internal_to_external_.begin(), internal_to_external_.end(), [&external](auto& p) {
-			if (p.second == &*external)
-				setStatus<dir>(p.first, external->priority().status());
-		});
+		auto internals{ externalToInternalMap().equal_range(&*external) };
```

Comment:
Doesn't `equal_range` require _sorted_ keys?

Reply:
No, [unordered_multimap::equal_range](https://en.cppreference.com/w/cpp/container/unordered_multimap/equal_range) is defined. [It does not keep the buckets sorted but keeps equal elements together](https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/hashtable.h#L1806).